File: analyze_price.py
import pandas as pd
import os
import sys

file_path = r'c:\Users\OLEG\Desktop\AZ\wood_canopy_calc\naves-calc\upload\naves\PRICE1.xlsm'

try:
    # Check if file exists
    if not os.path.exists(file_path):
        print(f"File not found: {file_path}")
        sys.exit(1)

    print(f"Analyzing {file_path}...")
    
    # Load Excel file (requires openpyxl)
    xl = pd.ExcelFile(file_path)
    
    print(f"Sheet names: {xl.sheet_names}")
    
    for sheet in xl.sheet_names:
        print(f"\n--- Sheet: {sheet} ---")
        df = xl.parse(sheet)
        print(f"Shape: {df.shape}")
        print("First 5 rows:")
        print(df.head().to_string())
        print("\nColumns:")
        print(list(df.columns))

except Exception as e:
    print(f"Error reading Excel file: {e}")

File: analyze_price.js
const XLSX = require('xlsx');
const path = require('path');
const fs = require('fs');

const filePath = path.join(__dirname, 'naves-calc', 'upload', 'naves', 'PRICE1.xlsm');

console.log(`Analyzing ${filePath}...`);

if (!fs.existsSync(filePath)) {
    console.error(`File not found: ${filePath}`);
    process.exit(1);
}

try {
    const workbook = XLSX.readFile(filePath);
    console.log(`Sheet names: ${workbook.SheetNames.join(', ')}`);

    workbook.SheetNames.forEach(sheetName => {
        console.log(`\n--- Sheet: ${sheetName} ---`);
        const worksheet = workbook.Sheets[sheetName];

        // Convert to JSON to see data
        const data = XLSX.utils.sheet_to_json(worksheet, { header: 1 }); // array of arrays

        if (data.length === 0) {
            console.log("Empty sheet");
        } else {
            console.log(`Rows: ${data.length}`);
            console.log("First 5 rows:");
            data.slice(0, 5).forEach((row, index) => {
                console.log(`Row ${index}: ${JSON.stringify(row)}`);
            });
        }
    });

} catch (error) {
    console.error(`Error reading Excel file: ${error.message}`);
}
